orchestrator-salt/src/main/resources/salt/salt/postgresql/scripts/init_db_remote.sh

```diff
@@ -6,18 +6,23 @@
 
 
 echo "Check if database already exists for {{ service }}"
-PGPASSWORD={{ values['remote_admin_pw'] }} psql --host={{ values['remote_db_url'] }} --port={{ values['remote_db_port'] }} --username={{ values['remote_admin'] }} -v "ON_ERROR_STOP=1" -lqt | awk {'print $1'} | grep -qw {{ values['database'] }}
+PGPASSWORD={{ values['remote_admin_pw'] }} psql --host={{ values['remote_db_host'] }} --port={{ values['remote_db_port'] }} --username={{ values['remote_admin'] }} -v "ON_ERROR_STOP=1" -lqt | awk {'print $1'} | grep -qw {{ values['database'] }}
 result=$?
 
 set -e
 if [[ $result -eq 0 ]]; then
     echo "Database already exists for {{ service }}, skipping initialization."
 else
+    username="{{ values['user'] }}"
+    username="${username%%@*}"
+    admin_username="{{ values['remote_admin'] }}"
+    admin_username="${admin_username%%@*}"
     echo "Create remote database and user for service {{ service }}"
-    PGPASSWORD={{ values['remote_admin_pw'] }} createdb --host={{ values['remote_db_url'] }} --port={{ values['remote_db_port'] }} --username={{ values['remote_admin'] }} {{ values['database'] }}
-    echo "CREATE USER {{ values['user'] }} WITH PASSWORD '{{ values['password'] }}';" | PGPASSWORD={{ values['remote_admin_pw'] }} psql --host={{ values['remote_db_url'] }} --port={{ values['remote_db_port'] }} --username={{ values['remote_admin'] }} -v "ON_ERROR_STOP=1" {{ values['database'] }}
-    echo "GRANT ALL PRIVILEGES ON DATABASE {{ values['user'] }} TO {{ values['database'] }};" | PGPASSWORD={{ values['remote_admin_pw'] }} psql --host={{ values['remote_db_url'] }} --port={{ values['remote_db_port'] }} --username={{ values['remote_admin'] }} -v "ON_ERROR_STOP=1" {{ values['database'] }}
-    echo "ALTER SCHEMA public OWNER TO {{ values['user'] }};" | PGPASSWORD={{ values['remote_admin_pw'] }} psql --host={{ values['remote_db_url'] }} --port={{ values['remote_db_port'] }} --username={{ values['remote_admin'] }} -v "ON_ERROR_STOP=1" {{ values['database'] }}
+    PGPASSWORD={{ values['remote_admin_pw'] }} createdb --host={{ values['remote_db_host'] }} --port={{ values['remote_db_port'] }} --username={{ values['remote_admin'] }} {{ values['database'] }}
+    echo "CREATE USER $username WITH PASSWORD '{{ values['password'] }}';" | PGPASSWORD={{ values['remote_admin_pw'] }} psql --host={{ values['remote_db_host'] }} --port={{ values['remote_db_port'] }} --username={{ values['remote_admin'] }} -v "ON_ERROR_STOP=1" {{ values['database'] }}
```

Comment:
Don't we need login role for this user? Apparently not, as the old code had no login, either, I just do not understand why.

Reply:
When you say "CREATE USER", the login attribute is automatically added.

https://www.postgresql.org/docs/9.1/sql-createuser.html